[lldb] Enable auto-detection of Swift/C++ interop

augusto2112 opened the pull request:
rdar://97610458

augusto2112 commented:
@swift-ci test

adrian-prantl commented on lldb/include/lldb/lldb-private-enumerations.h:
```diff
@@ -170,6 +170,14 @@ typedef enum SwiftModuleLoadingMode {
 } SwiftModuleLoadingMode;
 
+// BEGIN SWIFT
+enum EnableCxxInteropOpts {
```
Should we call this Enable**Swift**Cxx?

adrian-prantl commented on lldb/source/Core/Module.cpp:
```diff
@@ -1202,6 +1202,63 @@ void Module::ReportWarningToolchainMismatch(
+    // libraries shipped with Swift are written in C++.
+    auto name = GetFileSpec().GetFileNameStrippingExtension().GetStringRef();
+    if (name.startswith("libswift"))
+      return false;
```
This a per-module function, so this would misfire for libSwiftCore.dSYM, which generally only people manually building their own Swift compiler would even have. Since developers might legitimately name libraries `libswift*` maybe it's better to accept a false positive inside the standard library for those few people that have debug info for it?

adrian-prantl commented on lldb/source/Target/Target.cpp:
```diff
@@ -2860,6 +2860,42 @@ bool Target::IsSwiftREPL() {
+    for (size_t mi = 0; mi != num_images; ++mi) {
+      ModuleSP module_sp = GetImages().GetModuleAtIndex(mi);
+      if (module_sp->IsSwiftCxxInteropEnabled()) {
+        m_is_swift_cxx_interop_enabled = eLazyBoolYes;
```
Ah I see. The problem is that we merge the info from all modules.

adrian-prantl reviewed (approved):
Can you make sure this is printed in SwiftASTContext::LogConfiguration?

augusto2112 commented:
@swift-ci test

augusto2112 commented:
@swift-ci test macOS

augusto2112 commented:
@swift-ci test

augusto2112 commented:
@swift-ci test

augusto2112 commented:
@swift-ci test

augusto2112 commented:
@swift-ci test

augusto2112 commented:
https://github.com/apple/swift/pull/64958

@swift-ci test

augusto2112 committed:
[lldb] Implement SymbolFile::GetCompileOptions

Implement SymbolFile::GetCompileOptions, which returns a map from
compilation units to compilation arguments associated with that unit.

Differential Revision: https://reviews.llvm.org/D147748

(cherry picked from commit 19d969e340c9e1b5a83ad5220ba0875393df71e2)

augusto2112 committed:
[lldb] Enable auto-detection of Swift/C++ interop

rdar://97610458

augusto2112 commented:
@swift-ci test

augusto2112 commented:
https://github.com/apple/swift/pull/64958

@swift-ci test
